### ISSUE TITLE
Python 3

### ISSUE THREAD

**Issue by skolsuper** [hendrix/hendrix#113]
Is this library Python 3 compatible?  If not, how much work do you think it would be to port?

**Comment by JamesTFarrington** [comment]
@jMyles can probably give you a better answer on this, but I believe there is an issue with twisted's python 3 compatibility. We plan on moving to Python 3 when that is resolved.

**Comment by jMyles** [comment]
As I understand it, we're just waiting for this ticket, which has been submitted for review and will land any day now (hopefully this week):

https://twistedmatrix.com/trac/ticket/7993

If there *are* any other Python 3 issues following the application of this patch, my sense is that they'll be fairly easy to handle.

**Comment by pellagic-puffbomb** [comment]
I figured I would let you guys know that I was able to get Hendrix working on Python3.4 by installing the following versions:

```
git+git://github.com/twisted/twisted.git@trunk
git+git://github.com/hendrix/hendrix.git@master
```

And also by making the code changes that may be seen here: https://github.com/hendrix/hendrix/compare/master...pellagic-puffbomb:py3compat

One other thing to note: I had to bump the dependency on `pychalk` to 1.1.0.

Caveats and Things: In case anyone is dropping by and thinking about following my lead, I would offer the following:

- I have not tested this on Python2.
- I have only successfully run `hx start` (and had a confusing error that actually had to do with port 8000 being in use already, so I was thinking about working on some of the error messages here).
- When I ran tests, I got the following result:

```
FAILED (errors=7, successes=27)
```

A handful of these are `str`/`bytes` problems and then there was a twisted import issue.

**Comment by edwardotis** [comment]
+1 for Python 3 compatibility

**Comment by tjtimer** [comment]
what about this error?

$ hx start
Traceback (most recent call last):
  File "/var/www/tjtimer/.venv/matu/bin/hx", line 3, in <module>
    from hendrix.ux import main
  File "/var/www/tjtimer/.venv/matu/lib/python3.4/site-packages/hendrix/ux.py", line 231
    except Exception, Argument:
                    ^
SyntaxError: invalid syntax


using python 3.4 and Django 1.8

**Comment by citruspi** [comment]
@tjtimer 

That's just a change in the supported syntax for catching exceptions in Python.

In Python 2 you could use:

```python
except Exception, Argument:
    print(Argument)
```
or
```python
except Exception as Argument:
    print(Argument)
```

In Python 3 only the latter version is supported.

**Comment by jMyles** [comment]
Yeah, that's an obvious blunder.  Sorry about that.

@citruspi, care to turn your observation into a Pull Request? :-)

**Comment by citruspi** [comment]
@jMyles Already on it ;)

I just grep-ed the source for other similar errors, and I'll submit a PR this evening.

**Comment by jMyles** [comment]
Closed in #148
